### PR TITLE
Fix links and clarify confusing documentation

### PR DESCRIPTION
**Pull request by celestebancos**
I hope all my new links are correct — I couldn't find a way to test them locally.

### CODE REVIEW

**Comment by celestebancos** [comment]
Thank you, @dylburger!  I checked the Vercel preview and the images are still broken but all the other fixes are looking good. I'm not sure what the image links need to be to get the images to display.

I tried getting the preview running locally so I could test different image links but ran into issues running `yarn install` and `yarn docs:dev`:

```
C:\Users\celes\pipedream>yarn install
yarn install v1.22.10
info No lockfile found.
warning package-lock.json found. Your project contains lock files generated by tools other than Yarn. It is advised not to mix package managers in order to avoid resolution inconsistencies caused by unsynchronized lock files. To clear this warning, remove package-lock.json.
[1/4] Resolving packages...
warning lerna > @lerna/version > temp-write > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning lerna > @lerna/bootstrap > @lerna/run-lifecycle > npm-lifecycle > node-gyp > request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
warning lerna > @lerna/bootstrap > @lerna/run-lifecycle > npm-lifecycle > node-gyp > request > uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
warning lerna > @lerna/bootstrap > @lerna/symlink-dependencies > @lerna/create-symlink > @zkochan/cmd-shim > mkdirp-promise@5.0.1: This package is broken and no longer maintained. 'mkdirp' itself supports promises now, please switch to that.
warning lerna > @lerna/bootstrap > @lerna/run-lifecycle > npm-lifecycle > node-gyp > request > har-validator@5.1.5: this library is no longer supported
warning lerna > @lerna/create > globby > fast-glob > micromatch > snapdragon > source-map-resolve > resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
warning lerna > @lerna/create > globby > fast-glob > micromatch > snapdragon > source-map-resolve > urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
[2/4] Fetching packages...
error eslint-visitor-keys@3.0.0: The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "12.18.3"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

C:\Users\celes\pipedream>yarn docs:dev
yarn run v1.22.10
error Command "docs:dev" not found.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by dylburger** [comment]
@celestebancos you'll need to `cd docs` before running `yarn install` and `yarn docs:dev` - the docs app actually lives in that directory.

**Comment by celestebancos** [comment]
@dylburger I got the preview working locally and I fixed the image links. Everything looks good on my end, though I see that one of the Vercel previews has failed here. I'm not able to see the details on that so please let me know if there are any changes I need to make.

**Comment by dylburger** [comment]
@celestebancos the Vercel build failed because it doesn't have access to the images directory in the root of the repo, just the content in the `docs/` directory:

```
Module not found: Error: Can't resolve '../../../../images/image-20210326151930885.png' in '/vercel/path1/docs/components/guidelines'
```

Would you mind moving the image into the `docs/components/guidelines/images` directory and referencing that image in the Markdown, instead?

Reviewing the rest of the PR now.

File: docs/docs/components/api/README.md [hunk 1]
```diff
@@ -202,7 +202,7 @@ The event lifecycle applies to deployed components. Learn about the [component l
 
 ### Diagram
 
-![./image-20200819210516311](images/image-20200819210516311.png)
+![./image-20200819210516311](../../../../images/image-20200819210516311.png)
```
Comment (dylburger) [inline docs/docs/components/api/README.md:205]: See other comment in the "Conversation" section

File: docs/docs/components/guidelines/README.md [hunk 2]
```diff
@@ -258,16 +258,16 @@ out](https://pipedream.com/community/c/dev/11).
 
 ##### Prop Definitions
 
-Whenever possible, reuse existing [prop
-definitions](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#prop-definitions-example).
+Whenever possible, reuse existing [prop definitions](../api/#prop-definitions-example).
```
Comment (dylburger) [inline docs/docs/components/guidelines/README.md:261]: Relative links should be relative to the docs root for clarity, so this is preferred:

```
[prop definitions](/components/api/#prop-definitions-example)
```

File: docs/docs/components/guidelines/README.md [hunk 3]
```diff
@@ -258,16 +258,16 @@ out](https://pipedream.com/community/c/dev/11).
...
+
 If a prop definition does not exist and you are adding an app-specific prop that
 may be reused in future components, add it as a prop definition to the app file.
 Prop definitions will also be surfaced for apps the Pipedream marketplace.
 
 ##### Methods
 
 Whenever possible, reuse
-[methods](https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#methods)
+[methods](../api/#methods)
```
Comment (dylburger) [inline docs/docs/components/guidelines/README.md:270]: See comment above

File: docs/docs/components/guidelines/README.md [hunk 4]
```diff
@@ -373,13 +373,13 @@ Examples:
 - The async option to select an Airtable Base is self-explanatory so includes no
   description:
 
-  ![image-20210326151557417](images/image-20210326151557417.png)
+  ![image-20210326151557417](../../../../images/image-20210326151557417.png)
```
Comment (dylburger) [inline docs/docs/components/guidelines/README.md:376]: You'll also need to move these images into the local `images/` subdirectory (see main comment on why the Vercel build failed)

File: docs/docs/components/guidelines/README.md [hunk 5]
```diff
@@ -373,13 +373,13 @@ Examples:
...
 
 - The “Search Term” prop for Twitter includes a description that helps the user
   understand what values they can enter, with specific values highlighted using
   backticks and links to external content.
 
-  ![image-20210326151706682](images/image-20210326151706682.png)
+  ![image-20210326151706682](../../../../images/image-20210326151706682.png)
```
Comment (dylburger) [inline docs/docs/components/guidelines/README.md:382]: You'll also need to move these images into the local `images/` subdirectory (see main comment on why the Vercel build failed)

File: docs/docs/components/guidelines/README.md [hunk 6]
```diff
@@ -391,7 +391,7 @@ connect their account and enter a search term. The remaining fields are optional
 for users who want to filter the results, but they do not require any action to
 activate the source:
 
-![image-20210326151930885](images/image-20210326151930885.png)
+![image-20210326151930885](../../../../images/image-20210326151930885.png)
```
Comment (dylburger) [inline docs/docs/components/guidelines/README.md:394]: You'll also need to move these images into the local `images/` subdirectory (see main comment on why the Vercel build failed)

File: docs/docs/components/quickstart/nodejs/sources/README.md [hunk 7]
```diff
@@ -495,17 +495,19 @@ let axios = require("axios");
...
+},
 ```
 
 **IMPORTANT: The CLI will prompt you to select a connected account (or connect a new one) when you deploy (or update) this component.**
 
 > **Note:** The value for the `app` property is the name slug for the app in Pipedream. This is not currently discoverable, but it will be in the near future. For the time being, if you want to know how to reference an app, please reach out on our public Slack.
 
-Finally, we'll update the `run()` method to fetch issues from Github using `axios` and emit them. Notice that we're passing the `oauth_access_token` in the authorization header by referencing the app prop `this.github.$auth.oauth_access_token`. Also, similar to the RSS example, it's important that you use the `pddemo/demo` repo so you can test the next dedupe strategy.
+Finally, we'll update the `run()` method to fetch issues from Github using `axios` and emit them. Notice that we're passing the `oauth_access_token` in the authorization header by referencing the app prop `this.github.$auth.oauth_access_token`. Also, similar to the RSS example, it's important that you use `https://api.github.com/repos/pddemo/demo/issues` as the repo so you can test the next dedupe strategy.
```
Comment (dylburger) [inline docs/docs/components/quickstart/nodejs/sources/README.md:510]: I see why you made this change, but technically `pddemo/demo` is still the name of the repo, so I think we should keep that part as-is. Maybe modify to something like:

> Also, similar to the RSS example, it's important that you use the `pddemo/demo` repo so you can test the next dedupe strategy (see the `url` below).

to clarify that the reader should pay attention to the URL?

**Review by dylburger — CHANGES_REQUESTED** [review]
Just some small changes

File: docs/docs/components/quickstart/nodejs/sources/README.md (hunk 7 above)
Comment (celestebancos) [reply docs/docs/components/quickstart/nodejs/sources/README.md]: I think I was thrown off by the specific reference to the RSS example. I was looking at pddemo/demo and thinking, "Have I seen this before? Was this in the last example?" Reading further to the sample code cleared it up but there was definitely a moment of confusion there!

How about this wording?

> Again, it's important that you stick with the pddemo/demo repo shown in the below example so you can test the next dedupe strategy.

Comment (dylburger) [reply docs/docs/components/quickstart/nodejs/sources/README.md]: I like it!

Comment (celestebancos) [reply docs/docs/components/quickstart/nodejs/sources/README.md]: Great! Everything should be good to go then.
